### ISSUE TITLE
rstrnt-sync-* apparently not compatible with tmt

### ISSUE THREAD

**Issue by happz** [teemtee/tmt#2749]
When called for the first time, `rstrnt-sync-set` spawns a background process that listens on TCP & Unix sockets for incoming sync changes. However, when a test finishes, tmt & SSH cooperate to avoid leaving leftover processes on the guest, and this means the said background process would be killed. Probably quite strongly, as it does not clean up the Unix socket, and any subsequent call to `rstrnt-sync-set` on the same guest, e.g. from a later test, will get stuck on trying to connect to the now dead background sync manager process.

What to do with this situation? Do we want to provide compatible aliases? Or write a note & recommend other libraries?

**Comment by thrix** [comment]
@happz we want to strive for 100% compatibility with Beaker. So we should make it work.

We might want to provide our own `rstrnt-sync-set` that makes sure the background process correctly handles the signal handling and exists gracefully on such an event, cleaning up the leftover Unix socket.

**Comment by thrix** [comment]
notes from the tmt hacking discussion:

Can Jishi answer how many tests are affected by this, are there a lot of them?

**Comment by kkaarreell** [comment]
According to restraint docs
```
States are scoped to the current task. That is, states set by the current task will have no effect in subsequent tasks.
```
Even if we fix the process killing issue, does it mean that `tmt` would also mimic restraint behavior with respect to the above to avoid conflicts across tests?

**Comment by linup2011** [comment]
> notes from the tmt hacking discussion:
> 
> Can Jishi answer how many tests are affected by this, are there a lot of them?

sorry about the late response, I didn't notice about this.
many multi-host tests are using the rhts-sync-*, if it can't be fixed, we need to replace the rhts-sync-* with lib that is compatible with tmt and testing-farm

**Comment by psss** [comment]
@linup2011, could you please have a look at the alternatives proposed in the following section?

* https://tmt.readthedocs.io/en/stable/guide.html#synchronization-libraries

I'm afraid we won't be able to cover all the `restraint` features, specifically `rstrnt-sync-*` scripts might be out of scope.

**Comment by linup2011** [comment]
yes, I've tried sync_set and sync_wait in https://gitlab.cee.redhat.com/kernel-qe/kernel/-/blob/master/networking/common/lib/lib_nc_sync.sh?ref_type=heads, and also [ControlFlow](https://github.com/beakerlib/ControlFlow/pull/7), they can work in testing-farm

**Comment by therazix** [comment]
@linup2011 Thanks for confirmation.

Summary from hacking session: Document that implementing the sync command is out-of-scope for tmt. There are alternatives mentioned in the documentation.
